sanitizers/detectors/sql_injection.go

```diff
@@ -19,29 +19,31 @@ var syntaxErrors = []*regexp.Regexp{
 }
 
 type SQLInjection struct {
-	id    int    // numeric identifier to distinguish between the detectors for the various call sites
-	query string // the SQL query that has been executed
-	err   error  // the error resulting from executing the SQL query
+	id        int    // numeric identifier to distinguish between the detectors for the various call sites
+	query     string // the SQL query that has been executed
+	err       error  // the error resulting from executing the SQL query
+	faultType string // Fault type passed along to the Reporter
```

Comment:
The detectors return specialized errors they discover a bug. In the case of SQL injection, we return an `SQLInjectionError`, which has the same information that `faultType` would convey. I think it would be cleaner if callers then call `err.Error()` to get the error string representing the fault type.